modules/packages/Sort.chpl

```diff
@@ -591,55 +591,65 @@ proc binaryInsertionSort(Data: [?Dom] ?eltType, comparator:?rec=defaultComparato
  */
 proc mergeSort(Data: [?Dom] ?eltType, minlen=16, comparator:?rec=defaultComparator) {
   chpl_check_comparator(comparator, eltType);
-  _MergeSort(Data, minlen, comparator);
+  _MergeSort(Data, Dom.low, Dom.high+1, minlen, comparator);
 }
 
-private proc _MergeSort(Data: [?Dom], minlen=16, comparator:?rec=defaultComparator)
+private proc _MergeSort(Data: [?Dom], lo:int, hi:int, minlen=16, comparator:?rec=defaultComparator)
   where Dom.rank == 1 {
-  const lo = Dom.dim(1).low;
-  const hi = Dom.dim(1).high;
-  if hi-lo < minlen {
-    insertionSort(Data, comparator);
+  if (hi-lo < minlen) {
+    var sorted = Data[lo..(hi-1)];
+    insertionSort(sorted, comparator=comparator);
```

Comment:
After PR #12452, there is also a shellSort available. It's only noteworthy because it accepts a start and end position (including the end position) and so avoids the overhead of creating an array slice on the many small sort subproblems.

I think it'd be reasonable to use shellSort here or to adjust insertionSort to accept start and end indices (again, including the end position) to avoid the slice & improve performance.